### DISCUSSION TITLE
How to add new rows?

### DISCUSSION THREAD

**Discussion by hmanaog** [TanStack/table#2537] [General] [answered]
Hi, 
I started using react-table in my project a week ago and was able to understand the process bit by bit. The examples helped a lot.

Now, my goal is to be able to add new data to the current table. Like having an empty input field where I can put new data and an add button to have the new data inserted in the current table. I went thru all the examples of React-Table and I didn't find any.  Maybe someone has done this already and can guide me into this or like an example/demo code that I can study? I would really appreciate any guidance.

**Comment by gargroh** [comment, marked as answer]
You need to update `data` and react-table will re-render to display the same

**Reply by hmanaog** [reply]
Hi @gargroh, thank you for the reply. I was able to solve it with your hint.
This link also helped me understand the re-rendering of the react-table: https://github.com/tannerlinsley/react-table/issues/740

Thank you!

**Reply by sovetski** [reply]
it does not help on how to do it in real world example with the code, is it a best practice to store data in useState?

**Reply by cp0153** [reply]
Yes.
